File: common/map_sdk/datamodel/src/common/log.cpp
// Copyright 2024 Tencent Inc. All rights reserved.
//

#include "common/log.h"
#include <cstdarg>
#include <cstdio>
#include <iostream>

namespace txlog {

static int log_level = WARNING;

void setLevel(LogLevel level) { log_level = level; }

void trace(const std::string& err) {
  if (log_level <= ALL) std::cout << "[TRACE] " << err << std::endl;
}

void debug(const std::string& err) {
  if (log_level <= DEBUG) std::cout << "[DEBUG] " << err << std::endl;
}

void info(const std::string& err) {
  if (log_level <= INFO) std::cout << "[INFO] " << err << std::endl;
}

void warning(const std::string& err) {
  if (log_level <= WARNING) std::cout << "[WARNING] " << err << std::endl;
}

void error(const std::string& err) {
  if (log_level <= ERROR) std::cout << "[ERROR] " << err << std::endl;
}

void fatal(const std::string& err) {
  if (log_level <= FATAL) std::cout << "[FATAL] " << err << std::endl;
}

int printf(const char* format, ...) {
  constexpr int max_size = 10240;
  static char buffer[max_size] = {0};
  memset(buffer, 0, max_size);

  va_list va;
  va_start(va, format);
  int ret = vsnprintf(buffer, max_size, format, va);
  va_end(va);
  std::cout << buffer;
  return ret;
}
}  // namespace txlog

File: simcore/map_server/hadmap_server/hdserver/cloud/src/service/utils/log.go
/*
 Copyright (c) 2020 Tencent.com, Inc. All Rights Reserved

	Author: sbrucezhang(sbrucezhang@tencent.com)
	Date: 2020/10/14 20:24:11

*/
package utils

/*import (
	"fmt"
	"io"
	"os"
	"path"
	"runtime"
	"sync"

	"metric/config"
	//"filepath"
	//"metric/monitor"

	nested "github.com/antonfisher/nested-logrus-formatter"
	"github.com/sirupsen/logrus"
	formatter "github.com/zput/zxcTool/ztLog/zt_formatter"
	lumberjack "gopkg.in/natefinch/lumberjack.v2"
)

var (
	logLevels = map[string]logrus.Level{
		"TRACE": logrus.TraceLevel,
		"DEBUG": logrus.DebugLevel,
		"INFO":  logrus.InfoLevel,
		"WARN":  logrus.WarnLevel,
	}

	myFormatter *formatter.ZtFormatter = &formatter.ZtFormatter{
		CallerPrettyfier: func(f *runtime.Frame) (string, string) {
			filename := path.Base(f.File)
			return f.Function, fmt.Sprintf("%s:%d", filename, f.Line)
		},
		Formatter: nested.Formatter{
			HideKeys:     false,
			TrimMessages: true,
			FieldsOrder: []string{
				"component", "category", "filename",
				"network-type", "role", "target", "prefix", "address", "timeout-ms",
				"method", "url", "from", "request", "data", "statusCode", "response",
				"job-type", "job-id", "job-user", "job-stage", "job-is-retry",
				"error",
			},
		},
	}

	// Log is the logger object to be used
	Logger *logrus.Logger = func(f logrus.Formatter) *logrus.Logger {
		l := logrus.New()
		l.SetReportCaller(true)
		l.SetFormatter(f)
		l.SetLevel(logrus.DebugLevel)
		return l
	}(myFormatter)

	logger  *logrus.Entry //= Logger.WithField("Service", "daan_metric")
	logOnce sync.Once
)

func GetLogger() *logrus.Entry {

	if logger == nil {
		logOnce.Do(func() {
			logger = Logger.WithField("Service", "daan_metric")
		})
	}
	return logger
}

func InitLog(conf *config.Yaml) {
	_ = GetLogger()

	logRotate := &lumberjack.Logger{
		Filename:   conf.Log.FileName,
		MaxSize:    conf.Log.MaxSize, // MB
		MaxBackups: 0,
		MaxAge:     conf.Log.MaxAge, //days
		Compress:   false,
	}

	if conf.Log.AlsoToStderr {
		logWriter := io.MultiWriter(os.Stderr, logRotate)
		Logger.SetOutput(logWriter)
	} else {
		Logger.SetOutput(logRotate)
	}
	if level, ok := logLevels[conf.Log.Level]; ok {
		Logger.SetLevel(level)
	} else {
		fmt.Println("unknown log level")
		logger.Fatalf("Unrecognized log level %v", conf.Log.Level)
	}

	//LogRotateMonitor(conf)
}

*/
